app/client/keybase/keys.go

```diff
@@ -0,0 +1,326 @@
+		SecParam:   strconv.Itoa(secParam),
+		Hint:       hint,
+		CipherText: cipher,
+	}
+}
+
+// Generate new private ED25519 key and encrypt and armour it as a string
+// Returns a KeyPair struct of the Public Key and Armoured String
+func CreateNewKey(passphrase string) (KeyPair, error) {
+	privKey, err := poktCrypto.GeneratePrivateKey()
+	if err != nil {
+		return KeyPair{}, err
+	}
+
+	privArmour, err := encryptArmourPrivKey(privKey, passphrase)
+	if err != nil || privArmour == "" {
+		return KeyPair{}, nil
+	}
+
+	pubKey := privKey.PublicKey()
+	keyPair := NewKeyPair(pubKey, privArmour)
+
+	return keyPair, nil
+}
+
+// Generate new KeyPair from the hex string provided, encrypt and armour it as a string
+func CreateNewKeyFromString(privStr, passphrase string) (KeyPair, error) {
+	privKey, err := poktCrypto.NewPrivateKey(privStr)
+	if err != nil {
+		return KeyPair{}, err
+	}
+
+	privArmour, err := encryptArmourPrivKey(privKey, passphrase)
+	if err != nil || privArmour == "" {
+		return KeyPair{}, nil
+	}
+
+	pubKey := privKey.PublicKey()
+	keyPair := NewKeyPair(pubKey, privArmour)
+
+	return keyPair, nil
+}
+
+// Create new KeyPair from the JSON encoded privStr
+func ImportKeyFromJSON(jsonStr, passphrase string) (KeyPair, error) {
+	// Get Private Key from armouredStr
+	privKey, err := unarmourDecryptPrivKey(jsonStr, passphrase)
+	if err != nil {
+		return KeyPair{}, err
+	}
+	pubKey := privKey.PublicKey()
+	keyPair := NewKeyPair(pubKey, jsonStr)
+
+	return keyPair, nil
+}
+
+// Encrypt the given privKey with the passphrase, armour it by encoding the ecnrypted
+// []byte into base64, and convert into a json string with the parameters for unarmouring
+func encryptArmourPrivKey(privKey poktCrypto.PrivateKey, passphrase string) (string, error) {
+	// Encrypt privKey usign AES-256 GCM Cipher
+	saltBytes, encBytes, err := encryptPrivKey(privKey, passphrase)
+	if err != nil {
+		return "", err
+	}
+
+	// Armour encrypted bytes by encoding into Base64
+	armourStr := base64.StdEncoding.EncodeToString(encBytes)
+
+	// Create ArmouredKey object so can unarmour later
+	armoured := NewArmouredKey("scrypt", fmt.Sprintf("%X", saltBytes), "", armourStr)
+
+	// Encode armoured struct into []byte
+	js, err := json.Marshal(armoured)
+	if err != nil {
+		return "", err
+	}
+
+	return string(js), nil
+}
+
+// Encrypt the given privKey with the passphrase using a randomly
+// generated salt and the AES-256 GCM cipher
+func encryptPrivKey(privKey poktCrypto.PrivateKey, passphrase string) (saltBytes []byte, encBytes []byte, err error) {
+	// Use a default passphrase when none is given
+	if passphrase == "" {
+		passphrase = defaultPassphrase
+	}
+
+	// Get random bytes for salt
+	saltBytes = randBytes(16)
+
+	// Derive key for encryption, see: https://pkg.go.dev/golang.org/x/crypto/scrypt#Key
+	key, err := scrypt.Key([]byte(passphrase), saltBytes, n, r, p, klen)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// Encrypt using AES
+	privKeyBytes := privKey.String()
```

Comment:
Saw your comment here: https://github.com/pokt-network/pocket/pull/459#issuecomment-1402780661

Let's just make the var names appropriate.